app/assets/javascripts/orders.coffee

```diff
@@ -153,8 +161,8 @@ app.controller 'CheckoutController',['$scope', 'Product', 'Order', 'Address', 'C
   $scope.completeOrder = ->
     return if $scope.savingOrder
     $scope.savingOrder = true
-    $scope.order.state = 'placed'
-    $scope.order.$update()
+    $scope.ord.state = 'placed'
```

Comment:
the appController has a field named $scope.order and it was conflicting with this instance of $scope.order, hence the ord